tensorflow/compiler/xla/service/cpu/windows_compatibility.h

```diff
@@ -0,0 +1,35 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_COMPILER_XLA_SERVICE_CPU_WINDOWS_COMPATIBILITY_H_
+#define TENSORFLOW_COMPILER_XLA_SERVICE_CPU_WINDOWS_COMPATIBILITY_H_
+
+#include <math.h>
+
+extern "C" {
+
+void sincos(double x, double *sinv, double *cosv) {
```

Comment:
These should go into a .cc file with just the declarations in the header.  That way multiple `#include`s of `windows_compatibility.h` will not cause linker errors.

Reply:
> I'd avoid putting the _MSC_VER stuff here at all -- instead guard the entire contents of windows_compatibility.cc/h undef _MSC_VER.

Done

> Secondly, is _MSC_VER the right macro to use for this? Will it work do the right thing if we build with clang on windows?

Yes, it will work. `clang-cl.exe` tries very hard to pretend it is MSVC. I do like to build Tensorflow with Clang on Windows as well to see if it yields better performance.